src/libmongoc/tests/unified/result.c

```diff
@@ -361,10 +363,13 @@ result_check (result_t *result,
             error, "expected result, but got error: %s", result->error.message);
          goto done;
       }
-      if (!entity_map_match (em, expect_result, result->value, false, error)) {
-         test_set_error (error,
-                         "checking expectResult: %s",
-                         bson_val_to_json (expect_result));
+      if (!entity_map_match (em,
+                             expect_result,
+                             result->value,
+                             result->array_of_root_docs,
+                             error)) {
+         test_diagnostics_error_info ("checking expectResult: %s",
+                                      bson_val_to_json (expect_result));
```

Comment:
This ensures that the `entity_map_match` error with detailed info (e.g. path info) actually gets reported in the `test-libmongoc` output. Previously, it would have been discarded in favor of this less informative message.